github/repos.go

```diff
@@ -620,6 +620,23 @@ func (s *RepositoriesService) GetBranchProtection(owner, repo, branch string) (*
 	return p, resp, err
 }
 
+//GetProtectionRequest provides the ProtectionRequest struct to the caller
+func (s *RepositoriesService) GetProtectionRequest() *ProtectionRequest {
+	//TODO: Default values to be set, not thought about it yet
+	input := &ProtectionRequest{
+		RequiredStatusChecks: &RequiredStatusChecks{
+			IncludeAdmins: Bool(true),
+			Strict:        Bool(true),
+			Contexts:      &[]string{"continuous-integration"},
+		},
+		Restrictions: &BranchRestrictionsRequest{
+			Users: &[]string{"u"},
+			Teams: &[]string{"t"},
+		},
+	}
+	return input
+}
```

Comment:
This new method isn't needed. Everything it does can and should be done inside `TestRepositories_EditBranches` where it's currently being called.